### PR TITLE
Enable publishing capability when mutual SSL is selected as the only security scheme

### PR DESCRIPTION
**Pull request by isuruh15**
In publisher portal, user can publish a created API which doesn't have any application layer security and only having mutual SSL as security scheme. 

Fixes https://github.com/wso2/product-apim/issues/7522

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx [hunk 1]
```diff
@@ -194,7 +194,7 @@ class LifeCycleUpdate extends Component {
                     displayName,
                     disabled:
                         api.endpointConfig == null
-                        || api.policies.length === 0
+                        || (api.policies.length === 0 && !api.securityScheme.includes('mutualssl_mandatory'))
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx]: Shall we take this value from `apiSecurityConstants`

**Comment by kavishkafernando** [comment]
Shall we update the PR with the newly added checklist item and enable the Publish button only if the certificate, business plans and endpoint are provided provided?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx (hunk 1 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx]: Publish button should be disabled if the certificate is not provided as well if mutual SSL is enabled. We need to check for this condition.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycle.jsx [hunk 2]
```diff
@@ -30,6 +30,7 @@ import ApiContext from 'AppComponents/Apis/Details/components/ApiContext';
 import LifeCycleUpdate from './LifeCycleUpdate';
 import LifeCycleHistory from './LifeCycleHistory';
 
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycle.jsx]: Let's remove this extra line

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycle.jsx [hunk 3]
```diff
@@ -84,6 +86,7 @@ class LifeCycle extends Component {
         this.setState({ checkList });
     };
 
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycle.jsx]: Lets remove this extra line

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycle.jsx [hunk 4]
```diff
@@ -98,11 +101,14 @@ class LifeCycle extends Component {
 
         const promisedLcHistory = this.api.getLcHistory(id);
         // const promised_labels = this.api.labels();
-        Promise.all([promisedAPI, promisedLcState, promisedLcHistory])
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycle.jsx]: Let's remove this extra line

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycle.jsx [hunk 5]
```diff
@@ -98,11 +101,14 @@ class LifeCycle extends Component {
 
         const promisedLcHistory = this.api.getLcHistory(id);
         // const promised_labels = this.api.labels();
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycle.jsx]: Shall we remove this commented line if its not needed?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/CheckboxLabels.jsx [hunk 6]
```diff
@@ -120,22 +121,46 @@ export default function CheckboxLabels(props) {
...
+                                            id='Apis.Details.LifeCycle.CheckboxLabels.cert.provided'
+                                            defaultMessage='Certificate provided'
+                                        />
+                                    </Typography>
+                                    <Link to={'/apis/' + api.id + '/runtime-configuration'}>
+                                        <LaunchIcon style={{ marginLeft: '2px' }} color='primary' fontSize='small' />
+                                    </Link>
+                                </Grid>
+                            ) }
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/CheckboxLabels.jsx]: Shall we remove this extra line?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/CheckboxLabels.jsx [hunk 7]
```diff
@@ -120,22 +121,46 @@ export default function CheckboxLabels(props) {
                                 <LaunchIcon style={{ marginLeft: '2px' }} color='primary' fontSize='small' />
                             </Link>
                         </Grid>
-                        <Grid xs={12} className={classes.grid}>
-                            {isTierAvailable ? (
-                                <CheckIcon className={classes.iconTrue} />
-                            ) : (
-                                <CloseIcon className={classes.iconFalse} />
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/CheckboxLabels.jsx]: Shall we remove this extra line?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx [hunk 8]
```diff
@@ -152,13 +154,14 @@ class LifeCycleUpdate extends Component {
         this.updateLCStateOfAPI(apiUUID, action);
     }
 
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx]: Remove extra line

**Pull request merged**